Docs: Vitest plugin

kylegach opened the pull request:
## What I did

Add docs for the Vitest plugin

## Checklist for Contributors

### Testing

#### Manual testing

Sync locally running docs with this branch

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

<!-- greptile_comment -->

## Greptile Summary

This pull request introduces documentation for the new experimental Test runner with Vitest addon for Storybook, along with updates to existing testing documentation.

- Added `docs/writing-tests/test-runner-with-vitest.mdx` detailing installation, setup, and usage of the Vitest test runner
- Updated `docs/writing-tests/import-stories-in-tests/stories-in-unit-tests.mdx` to include Vitest examples for unit testing with Storybook stories
- Added `docs/_snippets/addon-vitest-install.md` and `docs/_snippets/addon-vitest-run-tests.md` with installation and test execution instructions
- Modified `docs/writing-tests/visual-testing.mdx` to include information about the new Visual Tests addon
- Adjusted sidebar ordering across multiple documentation files to accommodate the new Vitest content

<!-- /greptile_comment -->

kylegach committed:
WIP: Docs for Test runner with Vitest

- Includes sidebar order updates
    - Use non-zero-indexed orders

yannbf commented on docs/writing-tests/addon-vitest.mdx:
```diff
@@ -0,0 +1,346 @@
+  title: Test runner with Vitest
+---
+
+TK - Intro
```
We need to add (⚠️ **Experimental**) like in Playwright CT

yannbf commented on docs/writing-tests/addon-vitest.mdx:
```diff
@@ -0,0 +1,346 @@
+
+That command will do the following:
+
+1. Install and register the Vitest addon, which contains the plugin to run your stories as tests
+1. Inspect your project's Vite and Vitest setup
+    1. If Vite is not installed and you're using the [`nextjs` framework](../get-started/frameworks/nextjs.mdx), it will install and configure Vite for you, as well as `vite-plugin-storybook-nextjs` (necessary to have your Next.js components function in Vitest).
+    1. Otherwise, if Vite is not installed, it will stop and point you to these instructions to continue setting it up in your project.
+    1. If Vite is installed, it will then check for Vitest.
+        1. If Vitest is not installed, it will:
+            1. Install `vitest`, `@vitest/browser`, and `playwright`
+            1. Run `npx playwright install chromium` to install the Chromium browser engine
+            1. Create a Vitest config file (`vitest.config.ts`) and a Vitest setup file (`storybook.setup.ts`)
+        1. If Vitest is installed, it will stop and point you to these instructions to continue setting it up in your project.
```
It feels to me that we should just mention that the command will set up the plugin for the user. Hopefully the Storybook add command should be the place where the information is explained to the user as things are set up.

Then, we would have a section for manual steps that explains what to do:
- Make sure Vite and Vitest are configured and browser mode is used (point to vitest docs https://vitest.dev/guide/browser/)
- Install Next.js plugin if applicable (point to the docs https://github.com/storybookjs/vite-plugin-storybook-nextjs)
- Write set up file (point to portable stories docs)
- Write vitest config file

When it comes to the `npx playwright install chromium` command, Vitest doesn't mention it, I assume it's because it either does it for the user, or the user will get an error message from playwright telling them to run that using the correct package manager command. We should probably skip that in the docs

Feel free to use Vitest browser docs as reference:
https://vitest.dev/guide/browser/

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+            1. Create a Vitest config file (`vitest.config.ts`) and a Vitest setup file (`storybook.setup.ts`)
+        1. If Vitest is installed, it will stop and point you to these instructions to continue setting it up in your project.
+
+<If renderer="vue">
+	<Callout variant="info">
+		If your stories use template-based Vue components, you may need to [alias the `vue` module](https://vuejs.org/guide/scaling-up/tooling#note-on-in-browser-template-compilation) to resolve correctly in the Playwright CT environment. You can do this via the [`ctViteConfig` property](https://playwright.dev/docs/test-components#i-have-a-project-that-already-uses-vite-can-i-reuse-the-config):
+
+		<details>
+			<summary>Example Playwright configuration</summary>
+
+```ts
+// playwright-config.ts
+import { defineConfig } from '@playwright/experimental-ct-vue';
+
+export default defineConfig({
+  ctViteConfig: {
+    resolve: {
+      alias: {
+        vue: 'vue/dist/vue.esm-bundler.js',
+      },
+    },
+  },
+});
+```
+		</details>
+	</Callout>
+</If>
````
This should be related to Vite instead of Playwright

```ts
export default defineConfig({
  // ...
  resolve: {
    alias: {
      vue: 'vue/dist/vue.esm-bundler.js',
    },
  },
})
```

kylegach replied:
Does this also apply to https://storybook.js.org/docs/api/portable-stories/portable-stories-vitest?renderer=vue ?

yannbf replied:
@kylegach yes we could actually provide that as a solution for direct PS users too! Perhaps for PS in Playwright CT as well, via `use.ctViteConfig.plugins`

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+  <summary>Example Vitest setup file</summary>
+
+```ts title="storybook.setup.ts"
+TK
````
@kasperpeulen please share how things will look like from the sb add, it should match to make it more relatable to users that might end up coming from the CLI to the docs

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+  <summary>Example Vitest config file</summary>
+
+```ts title="vitest.config.ts"
+import { defineConfig, mergeConfig } from 'vitest/config'
````
@kasperpeulen please share how things will look like from the sb add, it should match to make it more relatable to users that might end up coming from the CLI to the docs

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+{/* TODO: Nextjs & SvelteKit examples */}
+```ts title="vitest.workspace.ts"
+import { defineWorkspace } from 'vitest/config'
+import { storybookTest } from '@storybook/experimental-addon-vitest/plugin'
````
@kasperpeulen please share how things will look like from the sb add, it should match to make it more relatable to users that might end up coming from the CLI to the docs

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+First, we run a command to build and publish Storybook. In this case, we'll use Chromatic. This gives us a URL to the published Storybook instance. We then pass that URL to the plugin configuration using an environment variable. Finally, we update the plugin configuration to use that environment variable in the `storybookUrl` option.
+
+```yaml
+TK
````
Perhaps this helps as reference: https://github.com/storybookjs/test-runner#1-running-against-deployed-storybooks-on-github-actions-deployment

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+```
+
+```js title="vitest.workspace.ts"
+process.env.SB_URL
````
```ts
      storybookTest({
        storybookScript: 'yarn storybook --ci',
        storybookUrl: process.env.SB_URL
      }),
```

if storybookUrl is undefined, it will use localhost:6006. for people who need other kind of urls:
```ts
      storybookTest({
        storybookScript: 'yarn storybook --ci',
        storybookUrl: process.env.SB_URL ?? 'http...'
      }),
```

yannbf commented on docs/writing-tests/addon-vitest.mdx:
```diff
@@ -0,0 +1,346 @@
+export default {
+  component: Button,
+  // 👇 Apply `test` tag to all stories in this file
+  tags: ['test'],
```
this is not needed in Storybook as `test` is added by default in all stories. Perhaps use another example? like `design` or something

yannbf commented on docs/writing-tests/addon-vitest.mdx:
````diff
@@ -0,0 +1,346 @@
+}
+```
+
+By default, the plugin will run all stories with the `test` tag. You can adjust this behavior by providing the [`tags` option](#tags) in the plugin configuration. This allows you to include, exclude, or skip stories based on their tags.
````
this feels a bit redundant, maybe we merge this on top with the text before the previous snippet? on line 177

yannbf commented on docs/writing-tests/addon-vitest.mdx:
```diff
@@ -0,0 +1,346 @@
+
+### How to ensure my tests can find assets in the public directory?
+
+If your stories use assets in the public directory and you're not using the default public directory location (`public`), you need to adjust the Vitest configuration to include the public directory. You can do this by providing the `publicDir` option in the Vitest configuration file.
```
for reference: https://vitejs.dev/config/shared-options.html#publicdir

yannbf commented on docs/writing-tests/addon-vitest.mdx:
```diff
@@ -0,0 +1,346 @@
+
+### Options
+
+#### `configDir`
```
If people have non standard locations for .storybook, this is quite important. This can happen when people change the name of the directory, and when they use monorepos or just restructure folder locations

yannbf reviewed (commented):
Good stuff! I hope @shilman can take a look soon once the PR is ready for review!

kylegach committed:
More updates

kylegach committed:
More updates

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,380 @@
+
+### Automatic
+
+Run the following command to install and configure the addon, which contains the plugin to run your stories as tests using Vitest:
```
We should mention to upgrade to storybook 8.3 first.

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,380 @@
+
+{/* TODO: Snippetize */}
+```sh
+npx storybook@latest add @storybook/experimental-addon-vitest
````
Didn't we had an agreement that `npx storybook add` is better for most command except upgrade and init? cc @shilman @valentinpalkovic have maybe more context here.

kylegach committed:
More polish

kylegach committed:
Merge branch 'next' into docs-addon-vitest

* next:
  Address feedback
  Address comments
  Typo
  Further updates
  Address feedback
  Remove TK
  Futher updates to #28716

kasperpeulen commented on docs/_snippets/addon-vitest-run-tests.md:
````diff
@@ -0,0 +1,11 @@
+```shell renderer="common" language="js" packageManager="npm"
+npm run test
+```
+
+```shell renderer="common" language="js" packageManager="pnpm"
+pnpm run test
+```
+
+```shell renderer="common" language="js" packageManager="yarn"
+yarn test
+```
````
this should be `yarn/pnpm/npm exec vitest` @kylegach 

the add command doesn't add a `test` script

kylegach replied:
I addressed this in the prose & snippets around this one.

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+  // This is the path to your existing Vitest config files
+  './vitest.config.ts', 
+  {
+    name: 'storybook',
```
we should add `extends: 'vite.config.ts'`

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+      provider: 'playwright',
+      headless: true,
+    },
+    // Disabling isolation is faster and similar to how tests are isolated in Storybook itself.
+    // Consider removing this, if you have flaky tests.
+    isolate: false,
```
@yannbf We should remove this, if we can not get the plugin to work with `isolate: false`

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+
+### CLI
+
+The plugin transforms your stories into real Vitest tests, so you run those tests just like you run any other Vitest tests in your project. Typically, you will have a `test` script in your `package.json` that runs Vitest. When you run that script, the addon will find and run your story-based tests. Here's an example of running your tests (in [watch mode](https://vitest.dev/guide/cli.html#vitest-w ...
```
@kylegach  Add something about adding a `test` script, if it doesn't exist.

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+      }),
+    ],
+    // Glob pattern to find story files
+    include: ['../src/**/*.stories.?(m)[jt]s?(x)'],
```
```suggestion
    include: ['src/**/*.stories.?(m)[jt]s?(x)'],
```

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,396 @@
+
+```ts title="vitest.config.ts"
+import { defineConfig, mergeConfig } from 'vitest/config'
+import viteConfig from '../vite.config'
````
```suggestion
import viteConfig from './vite.config'
```

kasperpeulen commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+      }),
+    ],
+    // Glob pattern to find story files
+    include: ['../src/**/*.stories.?(m)[jt]s?(x)'],
```
```suggestion
    include: ['src/**/*.stories.?(m)[jt]s?(x)'],
```

jonniebigodes commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+
+Default: `http://localhost:6006`
+
+The URL where Storybook is hosted. This is used for internal checks and to provide a link to the story in the test output on failures.
```
When you're able reword this to streamline it a bit to make a bit more clear that this is only for debugging/error states

kylegach committed:
Apply suggestions from code review

Co-authored-by: Kasper Peulen <kasperpeulen@gmail.com>

jonniebigodes commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+
+Tags to include, exclude, or skip. These tags are defined as annotations in your story, meta, or preview.
+
+- **`include`**: Stories with these tags will be tested
```
When you can add links to the tags documentation. Again, this is a new construct that users may or may not be aware.

jonniebigodes reviewed (changes requested):
Just did a first pass at this and left some feedback for you to look into when you can. Let me know and we'll go from there,

kylegach committed:
Further updates

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,396 @@
+])
+```
+
+If the same tag is in both the `include` and `exclude` arrays, the `exclude` behavior takes precedence.
````
I linked to the tags docs earlier in this section. Also linking it here seems redundant, especially because that docs page doesn't include anything about this `include`/`exclude` behavior, as that's part of the test runner (with or without Vitest).

I don't think this info is important enough to draw attention to it with a callout.

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+
+If the same tag is in both the `include` and `exclude` arrays, the `exclude` behavior takes precedence.
+
+## FAQ
```
Why? I debated back and forth on FAQ or Troubleshooting and landed on FAQ because it best fit the majority of the items.

A troubleshooting guide is oriented around solving problems people are having. Not all of these items fit that paradigm, while they all _can_ be worded as questions.

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,396 @@
+
+We recommend using Chromium, because it is most likely to best match the experience of a majority of your users. However, you can use other browsers by adjusting the [browser name in the Vitest configuration file](https://vitest.dev/config/#browser-name). Note that [Playwright and WebDriverIO support different browsers](https://vitest.dev/guide/browser/#browser-option-types).
+
+### How is this different from the previous test runner?
+
+The [previous test runner](./test-runner.mdx) requires a running Storybook instance to test your stories, because it visits each one, executes the play function, and listens for results. This plugin, however, transforms your stories into tests using Vite and portable stories, so it does not need to run Storybook to test your stories. Beyond that core difference, there are a few other distinctions ...
+
+Additionally, the previous test runner ran your stories as orchestrated tests in Jest, and that orchestration came with some complexity. By comparison, this plugin transforms your stories into real tests and then runs them using Vitest, which is simpler and more configurable.
+
+Finally, because of the simpler architecture and the use of Vitest, this plugin should be faster than the previous test runner for most projects. We'll do more benchmarking to quantify this in the future.
```
Sorry, I'm unclear on what you're asking for:

> Move this up...

What specifically is _this_ here? The whole section? Just the last paragraph?

> before the FAQ item

I think understanding what _this_ refers to will help, but I'm also not sure how to interpret this one

> we should avoid calling the original test runner as the previous one

Why? And how else should I refer to it?

jonniebigodes replied:
What I was referring to was removing this from the FAQ items, making it a section of itself, placing it above said section, and turning it into something as follows (based on the current content structure)

```md

### Debugging

While the plugin does not require Storybook to run when testing, you may still want to run Storybook to debug your tests......

****

### How is this different from the previous test runner?

The [previous test runner](./test-runner.mdx) requires a running Storybook instance to test your stories because it visits each one, executes the play function, and listens for results. This plugin, however, transforms your stories into tests using Vite and portable stories, so it does not need to run Storybook to test your stories. Beyond that core difference, there are a few other distinctions:

Additionally, the previous test runner ran your stories as orchestrated tests in Jest, and that orchestration came with some complexity. By comparison, this plugin transforms your stories into real tests and then runs them using Vitest, which is simpler and more configurable.

Finally, because of the simpler architecture and the use of Vitest, this plugin should be faster than the previous test runner for most projects. We'll do more benchmarking to quantify this in the future.

## FAQ

```



Applying this small change to the documentation increases the feature's discoverability and makes it easier for users to understand its purpose, key benefits, and why they should use it. As how it's currently documented, it's buried deep in the documentation (i.e., it's the last item on the FAQ).

kylegach replied:
Thanks for the additional details. I agree with this suggestion.

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,396 @@
+])
+```
+
+## Configuration
````
Above what?

It's currently placed here:

1. Intro (no heading)
2. Install and set up
3. Usage
4. Configuration

I strongly disagree it should go about any of the prior sections.

jonniebigodes replied:
What I was referring to was moving the configuration section upward after the installation, turning it:
- Intro
- Install and set up
- Configuration
- Usage


The rationale here is that this is an experimental feature that may require additional configuration on the user's part, which, placing this upwards, can get it up and running this feature more efficiently

kylegach replied:
Thank you for explaining further. I see your point, but disagree in this specific case. The configuration section only deals with configuring stories with tags, most commonly to exclude stories from being tested. I think running the tests (usage) should come before that, as it's likely only after that when folks will know which stories to exclude.

kylegach replied:
Maybe we should rename the section to `Configuring stories`?

kylegach committed:
Address feedback

kylegach committed:
Merge branch 'next' into docs-addon-vitest

* next: (33 commits)
  fix lint
  Clarify requirements of Next.js with Vite
  Next.js-Vite: Update next and vite-plugin-storybook-nextjs dependencies
  Addon Vitest: Fix tests potentially not existing in non-isolate mode
  CLI: Update spawn options in proxy.ts to support Windows
  Builder-Vite: Fix 'condition node never be used' warning
  Bump version from "8.3.0-alpha.9" to "8.3.0-alpha.10" [skip ci]
  Write changelog for 8.3.0-alpha.10 [skip ci]
  make prettier a normal dependency of core again. restore formatter code
  Use relative path from workspace to vite.config in extends path
  CLI: Fix dedent import in package managers
  Support JSX file extensions as well
  Addon Vitest: Fix postinstall file types
  Remove (disable) isolate option because it sometimes doesn't show stories
  add TODO comment
  make prettier optional
  improve meta file generation
  patch babel traverse types to be compatible with our dts bundler
  minimize changes
  minimize changes
  ...

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,396 @@
+```
+</details>
+
+## How it works
````
I ended up removing this section entirely in favor of a better, more thorough (but still more streamlined than this section as it was) intro.

kylegach committed:
Merge branch 'next' into docs-addon-vitest
* next:
  Address feedback
  add link to docs for how to set feature flags.
  improve migration notes for globals project
  Fix typo
  ConfigFile: Fix `as const satisfies` modifiers
  Bump version from "8.3.0-alpha.11" to "8.3.0-beta.0" [skip ci]
  Update docs/versions/next.json
  Add description to empty 8.3.0-beta.0 release
  Write changelog for 8.3.0-beta.0 [skip ci]
  Bump version from "8.3.0-alpha.10" to "8.3.0-alpha.11" [skip ci]
  Write changelog for 8.3.0-alpha.11 [skip ci]
  Re-order sidebar
  Rename interaction test -> component test

kylegach committed:
Fixes

kylegach committed:
Address comments and further changes

- Limit to react, vue, and svelte renderers
- `addon-vitest` -> `addon-test`
- `addon-vitest/plugin` -> `addon-test/vite-plugin`
- Add requirements
- Add more details and examples about framework plugins
- Add more detail about test setup file
- Move comparison to previous test runner into its own section
- Format snippets

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -19,53 +31,110 @@ Stories are tested in two ways: a smoke test to ensure it renders and, if a [pla
+</If>
+
 <If renderer="vue">
-	<Callout variant="info">
```
@yannbf — This (the `vue` alias) is now handled by `@storybook/vue3-vite/vite-plugin`, right? In other words, I was correct to remove this section?

yannbf replied:
Definitely!

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -19,53 +31,110 @@ Stories are tested in two ways: a smoke test to ensure it renders and, if a [pla
+  ```js title="vitest.config.ts"
+  import { defineConfig, mergeConfig } from 'vitest/config';
+  import { storybookTest } from '@storybook/experimental-addon-test/vite-plugin';
+  import { storybookNextjsPlugin } from '@storybook/experimental-nextjs-vite/vite-plugin';
````
@yannbf — How does this work for `@storybook/nextjs` users?

yannbf replied:
The same, they would need to install `@storybook/experimental-nextjs-vite` just for using `@storybook/experimental-nextjs-vite/vite-plugin`. It was a deliberate choice

kylegach committed:
Update heading

kylegach committed:
Merge branch 'next' into docs-addon-vitest

* next: (31 commits)
  add playwright eslint plugin, fix tests by rules
  fix tags e2e test
  fix urls in addon-test package.json
  use auto-retrying assertions in e2e tests
  renamed vitest addon import file name to 'test'
  Bump version from "8.3.0-beta.1" to "8.3.0-beta.2" [skip ci]
  Write changelog for 8.3.0-beta.2 [skip ci]
  fix package order in versions
  fix import
  fix
  add tests
  CLI: Handle Yarn PnP wrapper scenario when adding an addon
  cleanup
  Add type declaration for storybookTest in plugin module
  dont' use parameters for default backgrounds, as they are not overridable
  Fix type export for @storybook/experimental-nextjs-vite/vite-plugin
  don't use parameters for default viewports, as they are not overridable
  Nextjs-Vite: Re-export vite-plugin-storybook-nextjs
  refactor vite plugin export paths
  Bump version from "8.3.0-beta.0" to "8.3.0-beta.1" [skip ci]
  ...

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,502 @@
+If you have custom operations defined in [`viteFinal`](../api/main-config/main-config-vite-final.mdx) in your `.storybook/main.js|ts` file, you will need to translate those into the Vitest configuration. This is because the plugin does not use the Storybook Vite configuration.
+
+```ts
+TK - Is there a good example we could offer here?
````
Not sure, perhaps aliases?

```ts
// main.js
export default {
  viteFinal: async (viteConfig, { configType }) => {
    const { mergeConfig } = await import('vite');
  
    return mergeConfig(viteConfig, {
      resolve: {
        alias: {
          '@components': '/src/components',
          // ...
        }
      }
    })
  }
}
```

```ts
// vitest file
export default defineConfig({
  resolve: {
    alias: {
      '@components': '/src/components',
      // ...
    }
  },
  // ...
});
```

@JReinhold might have some other ideas

JReinhold replied:
I think the above alias is a good and concise example.

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+
+import * as projectAnnotations from './preview';
+
+const project = setProjectAnnotations(projectAnnotations);
```
it's easier if it's an array already so that users know they can add more. I made this change to the code generated by the addon as well
```suggestion
const project = setProjectAnnotations([projectAnnotations]);
```

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,502 @@
+beforeAll(project.beforeAll);
+```
+
+  The `setProjectAnnotations` function is part of the portable stories API, which is used to transform your stories into tests. For more details, see the [portable stories API documentation](../api/portable-stories/portable-stories-vitest.mdx#setprojectannotations).
````
```suggestion
  The `setProjectAnnotations` function is part of the portable stories API, which is used internally by the Vitest test runner to transform your stories into tests. For more details, see the [portable stories API documentation](../api/portable-stories/portable-stories-vitest.mdx#setprojectannotations).
```

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+  defineConfig({
+    plugins: [
+      storybookTest({
+        storybookScript: 'yarn storybook --ci',
```
might make sense to add a comment at the top saying that the user must alter the script based on their package.json. They could be using pnpm, npm, etc. Or maybe their script is not called `storybook` but rather `ui:dev` or whatever else

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+  viteConfig,
+  defineConfig({
+    plugins: [
+      storybookTest({
```
tricky, but if a user is setting up this manually they should know that the framework plugins section is important. This snippet doesn't show the potential framework plugin here, maybe people will miss it? Not sure what's the best way to communicate that here:
- A small note before the snippet
- A code comment with the URL to the framework plugin section in the docs
- A code comment mentioning "make sure to include necessary framework plugins (from sections above) if applicable"

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+        storybookScript: 'yarn storybook --ci',
+      }),
+    ],
+    test: {
+      // Glob pattern to find story files
```
```suggestion
    name: 'storybook',
    test: {
      // Glob pattern to find story files
```

yannbf commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+
+Transforming your stories into Vitest tests with the plugin also enables you to run and debug tests using Vitest [IDE integrations](https://vitest.dev/guide/ide.html). This allows you to run tests directly from your editor, such as VSCode and JetBrains IDE.
+
+[TK - Screenshot of VS Code]
```
I can assist but I think @jonniebigodes knows better on making consistent screenshots throughout the docs?

The feature is about showing the sidebar and check icons on the story file:
<img width="802" alt="image" src="https://github.com/user-attachments/assets/85d3d453-86e0-4ade-90b3-696b57fd883d">

This appears when you use the [vscode extension](https://vitest.dev/guide/ide.html) and set up the plugin

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
````diff
@@ -0,0 +1,502 @@
+
+ Storybook stories contain configuration defined in `.storybook/preview.js|ts`. To ensure that configuration is available to your tests, you can apply it in a Vitest setup file. Here's an example of how to do that:
+
+```ts title=".storybook/vitest.setup.ts"
````
Hmm... those are a mix of renderers (`vue3`, `svelte`) and frameworks (`sveltekit`, `nextjs`, `nextjs-vite`, `react-vite`). `sveltekit`, `nextjs`, and `nextjs-vite` are perhaps special cases. I think either the renderers should become frameworks (`vue3` and `svelte` should be `vue3-vite` and `svelte-vite`) or the frameworks should be renderers (`react-vite` should be `react`).

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+  <Callout variant="info">
+    Test runner with Vitest is currently only supported in [React](?renderer=react), [Vue](?renderer=vue) and [Svelte](?renderer=svelte) projects.
+
+    If you are using a different renderer, you can use the [previous test runner](./test-runner.mdx) to test your stories.
```
João had that feedback, too, but I don't know what else to call. Suggestions?

yannbf replied:
I'd relay that to @shilman, as it's an important product decision

kylegach commented on docs/writing-tests/test-runner-with-vitest.mdx:
```diff
@@ -0,0 +1,502 @@
+
+For some project setups, the `add` command may be unable to automate the plugin setup and ask you to complete additional setup steps. Here's what to do:
+
+1. Make sure Vite and Vitest are configured in your project.
+1. Configure Vitest to use [browser mode](https://vitest.dev/guide/browser/).
+1. Install the addon, `@storybook/experimental-addon-test`, in your project and [register it in your Storybook configuration](http://storybook.js.org/docs/addons/install-addons#manual-installation).
+1. Create a test setup file, `.storybook/vitest.setup.ts`. You can use the example setup file, below, as a guide
+1. Adjust your Vitest configuration to include the plugin(s) and reference the setup file. You can use the example configuration files, below, as a guide.
```
FYI, markdown renders a list with all `1.`s as a numbered list with 1, 2, 3...

kylegach committed:
Apply suggestions from code review

Co-authored-by: Yann Braga <yannbf@gmail.com>

kylegach committed:
Rename to 'Storybook Vitest plugin', and more

kylegach committed:
Merge branch 'next' into docs-addon-vitest

* next: (25 commits)
  Bump version from "8.3.0-beta.3" to "8.3.0-beta.4" [skip ci]
  Write changelog for 8.3.0-beta.4 [skip ci]
  rename vitest plugin entrypoint
  add narrow story to Sidebar
  upgrade @radix-ui/react-scroll-area
  Bump version from "8.3.0-beta.2" to "8.3.0-beta.3" [skip ci]
  Write changelog for 8.3.0-beta.3 [skip ci]
  fix lint
  only append addon to main.js if not already added
  final fixes
  Next.js: Update dependencies
  only install the necessary dependencies
  Next.js-Vite: Fix vite plugin exports
  use named import from dedent
  review improvements
  add explanation in sb add command
  add helpful links
  additional fixes
  Consistently naming the 'Storybook Test' addon
  CONSTANT_CASE for constants
  ...

kylegach committed:
Fix sidebar order
